src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -508,7 +509,14 @@ function transformConstraint(constraint, inArray) {
   if (typeof constraint !== 'object' || !constraint) {
     return CannotTransform;
   }
-
+  const transformFunction = inArray ? transformInteriorAtom : transformTopLevelAtom;
+  const transformer = (atom) => {
+    const result = transformFunction(atom);
```

Comment:
what!?  so how does this work? tranformFunction takes atom as an argument to line 512 replacing inArray?  no way.